cleanlab/dataset.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+
+"""Dataset Module
+Supports dataset-level and class-level automated quality, including finding which classes should
+be merged with other classes in your dataset, which classes should be removed,
+and which classes have the best label accuracy, and which have the worst labels.
+"""
+
+import numpy as np
+from count import estimate_joint, compute_confident_joint
+
+
+def find_similar_classes(
```

Comment:
Should we replace "similar" -> "commonly_confused" like we had in our cleanlab-studio report?

One reason is similar_classes just sounds like something I could get from a confusion matrix from model's predictions.

Reply:
I would think "confused" sounds more like it comes from a "confusion" matrix 🤣

Reply:
Ya that is indeed annoying as well... 

Also sorry I didn't even meant confusion matrix in my criticism of "similar" but rather all that's needed is model's pred_probs to find similar classes (can just measure correlation in pred_probs).

Alternatives names for this function could include:
`find_conflated_classes`
`find_swapped_classes`
`label_error_types`
`mislabeling_rates`

Reply:
how about `find_classes_to_merge` ?